ads/google/a4a/traffic-experiments.js

```diff
@@ -150,6 +140,27 @@ export function googleAdsIsA4AEnabled(win, element, experimentName,
   }
 }
 
+export function extractUrlExperimentId(win, element) {
+  const expParam = viewerForDoc(element).getParam('exp') ||
+    parseQueryString(win.location.search)['exp'];
+  if (!expParam) {
+    return null;
+  }
+  // Allow for per type experiment control with Doubleclick key set for 'da'
+  // and AdSense using 'aa'.  Fallbsck to 'a4a' if type specific is missing.
+  const expKeys = [
+    (element.getAttribute('type') || '').toLowerCase() == 'doubleclick' ?
+      'da' : 'aa',
+    'a4a',
+  ];
+  let arg;
+  let match;
+  expKeys.forEach(key => arg = arg ||
```

Comment:
It's not my code in THIS PR (meaning I wrote it previously).  I will be removing a4a soon and will clean it up then.